src/swupd_exit_codes.h

```diff
@@ -1,29 +1,36 @@
 #ifndef __INCLUDE_GUARD_SWUPD_ERROR_H
 #define __INCLUDE_GUARD_SWUPD_ERROR_H
 
-/* error codes to be used as swupd exit status */
+/* codes to be used as swupd exit status */
 
-#define EBUNDLE_MISMATCH 2      /* at least one local bundle mismatches from MoM */
-#define EBUNDLE_REMOVE 3	/* cannot delete local bundle filename */
-#define EMOM_LOAD 4		/* MoM cannot be loaded into memory (this could imply network issue) */
-#define ETYPE_CHANGED_FILE_RM 5 /* do_staging() couldn't delete a file which must be deleted */
-#define EDIR_OVERWRITE 6	/* do_staging() couldn't overwrite a directory */
-#define EDOTFILE_WRITE 7	/* do_staging() couldn't create a dotfile */
-#define ERECURSE_MANIFEST 8     /* error while recursing a manifest */
-#define ELOCK_FILE 9		/* cannot get the lock */
-#define ECURL_INIT 11		/* cannot initialize curl agent */
-#define EINIT_GLOBALS 12	/* cannot initialize globals */
-#define EBUNDLE_NOT_TRACKED 13  /* bundle is not tracked on the system */
-#define EMANIFEST_LOAD 14       /* cannot load manifest into memory */
-#define EINVALID_OPTION 15      /* invalid command option */
-#define ENOSWUPDSERVER 16       /* no net connection to swupd server */
-#define EFILEDOWNLOAD 17	/* file download problem */
-#define EBUNDLE_INSTALL 18      /* Cannot install bundles */
-#define EREQUIRED_DIRS 19       /* Cannot create required dirs */
-#define ECURRENT_VERSION 20     /* Cannot determine current OS version */
-#define ESIGNATURE 21		/* Cannot initialize signature verification */
-#define EBADTIME 22		/* System time is bad */
-#define EDOWNLOADPACKS 23       /* Pack download failed */
-#define EBADCERT 24		/* unable to verify server SSL certificate */
+typedef enum {
```

Comment:
++ for creating an enum for that. It's a lot better than returning just an int. 
But there's one thing I would change. We don't use a lot of typedefs in swupd code. I noticed that they use that for enums on telemetry.c and xattrs.c, but as we don't use that for structs I prefer to avoid using that for enums.

So I prefer:

    enum swupd_code {

And

    enum swupd_code my_function()

Reply:
my idea is that functions that need to return a documented code are defined to explicitly return a _swupd_code_, for example:

`swupd_code my_function() { ... }`

I think in order to do that we need to have the enum defined as a typedef.

Reply:
we can do it adding the prefix enum. take a look at src/swupd_curl_internal.h

Reply:
so do you prefer to use: `enum swupd_code my_func()` over `swupd_code my_func()`? If so, why? I'm not complaining just curious.